src/common/common.go

```diff
@@ -248,35 +248,85 @@ func parseOpenshiftVersionIntoMajorMinorZOnly(version string) string {
 }
 
 // HasValidvSphereCredentials returns true if the the platform is
-// vSphere and the install config overrides contains real
+// vSphere and the install config or equivalent overrides contains real
 // credential values and not placeholder values.
 // Deprecated credential fields are not considered valid.
-func HasValidvSphereCredentials(ctx context.Context, ic inventory_client.InventoryClient, log logrus.FieldLogger) bool {
+func HasValidvSphereCredentials(ctx context.Context, ic inventory_client.InventoryClient, kc k8s_client.K8SClient, log logrus.FieldLogger) bool {
 	cluster, callErr := ic.GetCluster(ctx, false)
+	if cluster != nil {
```

Comment:
Should not we check for invoker here? I believe service can be not reachable for period of time and i believe if invoker is agent there is not need for assisted service at all, right?

Reply:
The check to see whether there are valid vsphere credentials is only relevant when the invoker is agent. Perhaps I should have RemoveUnitializedTaint call HasValidvSphereCredentials when invoker == agent. Is this what you were suggesting?

In regards to your second question, when assisted-service cannot be reached, this function should then look at the install-config (in else clause) to see if valid credentials were specified there. If both assisted-service and kube-api cannot be reached to check credentials then the function should return false.

Reply:
Yep, i don't see why not agent code should actually run HasValidCredentials, it makes it not clear in non-agent flow

Reply:
Ok, I'll separate it out of non-agent flow.